### DISCUSSION TITLE
Using exception messages in production

### DISCUSSION THREAD

**Discussion by ianwremmel** [remix-run/remix#7301] [Q&A] [answered]
I have a set of exception classes extended from `Error` where the messages (and sometimes other properties, like the sentry trade id) are intended for consumption by end users. Of course, this works fine in development mode, but otherwise, `sanitizeError` replaces all the useful information with a generic error. 

Is there anywhere to hook in to customize `sanitizeError`? Alternatively, is there a generic spot to catch Errors before they get sanitized and instead convert them into Response objects?

**Comment by kiliman** [comment, marked as answer]
I don't think you can. In production, Remix always strips the Error before sending it to the client. The only chance your app has to access the Error is `handleError` export in *entry.server.tsx*, but this doesn't prevent the sanitization.

If you really need to send data to the client, you'll need to wrap in `try/catch` and send a new Response directly.

The other option is to `patch-package` and remove the santization.
